### PR TITLE
Ps ia welcome prompt to add first item

### LINKED ISSUE #7: 7. As a user, I want to see a welcoming prompt to add my first item if my list is empty to help me get oriented to how the app works.
## Summary

When a user starts a new list, they need a little guidance on how to get started. Think through what you'd want to see as a user and make some intentional user experience decisions that will guide our users in confidently adding their very first item. 

## Acceptance criteria
- [x] The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item

## Notes
- [The wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-empty.png) is a little short on the "welcoming" part of the story, but otherwise is functionally complete for the purposes of the AC.

### PR DESCRIPTION
**Pull request by Peacesandy**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

As a user, I want to see a welcoming prompt to add my first item if my list is empty to help me get oriented to how the app works. When a user starts a new list, they need a little guidance on how to get started. Think through what you'd want to see as a user and make some intentional user experience decisions that will guide our users in confidently adding their very first item.

## Related Issue

closes #7 

## Acceptance Criteria

- [ ] The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before

before#7

### After

promptauser

## Testing Steps / QA Criteria

As a user, you get a prompt message to add your first item, click the add item button to add your first item.

### CODE REVIEW

File: src/views/List.jsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { ListItem } from '../components';
-import { Navigate } from 'react-router-dom';
+import { Link, Navigate } from 'react-router-dom';
 import { useState, useEffect } from 'react';
+import { addItem } from '../api';
```
Comment (a-maffei) [inline src/views/List.jsx]: Since we're not using it, I imagine we can clean up the component and remove this import

File: src/views/List.jsx [hunk 2]
```diff
@@ -22,18 +23,31 @@ export function List({ data, listToken }) {
 		return <Navigate to="/" />;
 	}
 
+	//  const additemHandler = () => {
+	// 	return < Navigate to = '/add-item' />
+	//  }
+
```
Comment (a-maffei) [inline src/views/List.jsx]: Also here, we can clean up the component and remove this code we don't need.

**Review by a-maffei — APPROVED** [review]
@iblas1 @Peacesandy I left two suggestions, very small because the issue is very straightforward and the AC has been met. Well done!

**Comment by naura1835** [comment]
Everything looks great. Weldone

File: src/views/List.jsx [hunk 3]
```diff
@@ -22,18 +23,31 @@ export function List({ data, listToken }) {
...
-				<label htmlFor="search-input"> Filter items </label>
-				<input
-					type="search"
-					placeholder="Start typing here..."
-					name="search-item"
-					id="search-input"
-					onChange={handleChange}
-				/>
-			</form>
+			{data.length > 0 ? (
```
Comment (redapy) [inline src/views/List.jsx]: checking if the data exist first before accessing its length will avoid some possible errors in the future. 
You can either use the "and operator" `data && data.length > 0`, or a shorter version with the optional chaining operator  `data?.length  > 0`

**Review by redapy — APPROVED** [review]
well done, LGTM!!

**Review by naura1835 — APPROVED** [review]

File: src/views/List.jsx (hunk 2 above)
Comment (iblas1) [reply src/views/List.jsx]: All suggestions carried out. Thanks for your input

File: src/views/List.jsx (hunk 3 above)
Comment (iblas1) [reply src/views/List.jsx]: yes, we worried about this while writing the code. Thanks for showing us a cleaner way of doing this. Suggestion implemented.

**Pull request merged**